src/lib/components/barchartv2/utils.ts

```diff
@@ -39,61 +39,92 @@ export const getMaxValue = (data: { [key: string]: string | number }[]) => {
 };
 
 /**
- * Generates all days between start and end timestamps (inclusive)
+ * Generates time ranges between start and end timestamps based on the given interval
  * @param startTimestamp - Start timestamp in milliseconds
  * @param endTimestamp - End timestamp in milliseconds
- * @returns Array of timestamps for each day at midnight
+ * @param interval - Interval in milliseconds
+ * @returns Array of time ranges with start and end properties
  */
-const generateDayTimestamps = (
+const generateTimeRanges = (
   startTimestamp: number,
   endTimestamp: number,
-): number[] => {
-  const dayTimestamps: number[] = [];
-  const startDate = new Date(startTimestamp);
-  const endDate = new Date(endTimestamp);
+  interval: number,
+): { start: number; end: number }[] => {
+  const ranges: { start: number; end: number }[] = [];
+  if (!startTimestamp || !endTimestamp || !interval) {
+    return ranges;
+  }
+
+  let currentTimestamp = startTimestamp;
+  while (currentTimestamp <= endTimestamp) {
+    const rangeEnd = currentTimestamp + interval;
+
+    ranges.push({
+      start: currentTimestamp,
+      end: rangeEnd,
+    });
 
-  // Set to midnight to ensure consistent day boundaries
-  startDate.setHours(0, 0, 0, 0);
-  endDate.setHours(0, 0, 0, 0);
+    currentTimestamp += interval;
+  }
 
-  const currentDate = new Date(startDate);
+  return ranges;
+};
 
-  while (currentDate <= endDate) {
-    dayTimestamps.push(currentDate.getTime());
-    currentDate.setDate(currentDate.getDate() + 1);
+/**
+ * Formats a timestamp based on the interval
+ * @param timestamp - Timestamp in milliseconds
+ * @param interval - Interval in milliseconds
+ * @returns Formatted string
+ */
+const formatTimestamp = (timestamp: number, interval: number): string => {
+  const date = new Date(timestamp);
+
+  if (interval > 24 * 60 * 60 * 1000) {
+    return (
+      DAY_MONTH_FORMATER.format(date).replace(/[ ,]/g, '') +
+      ' ' +
+      TIME_FORMATER.format(date)
+    );
+  } else if (interval === 24 * 60 * 60 * 1000) {
+    // Daily or longer intervals - use day format
+    return DAY_MONTH_FORMATER.format(date).replace(/[ ,]/g, '');
+  } else if (interval >= 60 * 60 * 1000) {
+    // Hourly intervals - use hour format
+    const hours = TIME_FORMATER.format(date);
+    const day = DAY_MONTH_FORMATER.format(date).replace(/[ ,]/g, '');
+    return `${day} ${hours}`;
```

Comment:
For hourly intervals and minute intervals, maybe we could display the label only with hours. 
otherwise I feel too many repetition for day .